src/stream/src/executor_v2/project.rs

```diff
@@ -0,0 +1,209 @@
+// Copyright 2022 Singularity Data
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::fmt::{Debug, Formatter};
+
+use itertools::Itertools;
+use risingwave_common::array::column::Column;
+use risingwave_common::array::{DataChunk, StreamChunk};
+use risingwave_common::catalog::{Field, Schema};
+use risingwave_expr::expr::BoxedExpression;
+
+use super::{Executor, ExecutorInfo, SimpleExecutor, SimpleExecutorWrapper, StreamExecutorResult};
+use crate::executor::PkIndicesRef;
+use crate::executor_v2::error::StreamExecutorError;
+
+pub type ProjectExecutor = SimpleExecutorWrapper<SimpleProjectExecutor>;
+
+impl ProjectExecutor {
+    pub fn new(input: Box<dyn Executor>, exprs: Vec<BoxedExpression>, execuotr_id: u64) -> Self {
+        let info = input.info();
+
+        SimpleExecutorWrapper {
+            input,
+            inner: SimpleProjectExecutor::new(info, exprs, execuotr_id),
+        }
+    }
+}
+
+/// `ProjectExecutor` project data with the `expr`. The `expr` takes a chunk of data,
+/// and returns a new data chunk. And then, `ProjectExecutor` will insert, delete
+/// or update element into next operator according to the result of the expression.
+pub struct SimpleProjectExecutor {
+    info: ExecutorInfo,
+
+    /// Expressions of the current projection.
+    exprs: Vec<BoxedExpression>,
+}
+
+impl SimpleProjectExecutor {
+    pub fn new(input_info: ExecutorInfo, exprs: Vec<BoxedExpression>, executor_id: u64) -> Self {
+        let schema = Schema {
+            fields: exprs
+                .iter()
+                .map(|e| Field::unnamed(e.return_type()))
+                .collect_vec(),
+        };
+        Self {
+            info: ExecutorInfo {
+                schema,
+                pk_indices: input_info.pk_indices,
+                identity: format!("ProjectExecutor {:X}", executor_id),
+            },
+            exprs,
+        }
+    }
+}
+
+impl Debug for SimpleProjectExecutor {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        f.debug_struct("ProjectExecutor")
+            .field("exprs", &self.exprs)
+            .finish()
+    }
+}
```

Comment:
Seems unnecessary?

Reply:
I referred to [filter.rs](https://github.com/singularity-data/risingwave/blob/main/src/stream/src/executor_v2/filter.rs#L65-L71) and it implements the Debug trait.